cmd/collector/app/span_processor.go

```diff
@@ -180,23 +182,46 @@ func (sp *spanProcessor) processItemFromQueue(item *queueItem) {
 	sp.metrics.InQueueLatency.Record(time.Since(item.queuedTime))
 }
 
-func (sp *spanProcessor) addCollectorTags(span *model.Span) {
-	if len(sp.collectorTags) == 0 {
-		return
+func (sp *spanProcessor) dedupTags(span *model.Span) {
+	// Copy the collector tags to keep track of which of these need to be appended.
+	cTags := make(map[string]string, len(sp.collectorTags))
+	for k, v := range sp.collectorTags {
+		cTags[k] = v
 	}
-	dedupKey := make(map[string]struct{})
-	for _, tag := range span.Process.Tags {
-		if value, ok := sp.collectorTags[tag.Key]; ok && value == tag.AsString() {
-			sp.logger.Debug("ignore collector process tags", zap.String("key", tag.Key), zap.String("value", value))
-			dedupKey[tag.Key] = struct{}{}
+	var procTags []model.KeyValue
+	for _, sTag := range span.Process.Tags {
+		appendSpanTag := true
+		// If there's a tag key conflict.
+		if cTagVal, ok := sp.collectorTags[sTag.Key]; ok {
+			switch sp.useOnTagKeyConflict {
+			case "both":
+				// If both key and value are conflicting, there's no need to add the collector's tag.
+				if cTagVal == sTag.AsString() {
+					delete(cTags, sTag.Key)
+				}
+			case "collector":
+				appendSpanTag = false
+			case "span":
+				// If we choose the span's tag, then remove this conflicting tag from the collector tags.
+				delete(cTags, sTag.Key)
+			}
 		}
-	}
-	// ignore collector tags if has the same key-value in spans
-	for k, v := range sp.collectorTags {
-		if _, ok := dedupKey[k]; !ok {
-			span.Process.Tags = append(span.Process.Tags, model.String(k, v))
+		if appendSpanTag {
+			procTags = append(procTags, sTag)
 		}
 	}
+	// Append the remaining collector tags.
+	for k, v := range cTags {
+		procTags = append(procTags, model.String(k, v))
+	}
+	span.Process.Tags = procTags
+}
+
+func (sp *spanProcessor) addCollectorTags(span *model.Span) {
+	if len(sp.collectorTags) == 0 {
+		return
+	}
+	sp.dedupTags(span)
```

Comment:
this behavior should be identical to the agent's tags, so we should have a util class somewhere that stores the tags and adds them to span